src/osd/ReplicatedBackend.cc

```diff
@@ -870,10 +872,22 @@ void ReplicatedBackend::_do_pull_response(OpRequestRef op)
 	this,
 	m->get_priority());
     c->to_continue.swap(to_continue);
+    // Determine recovery cost
+    int cost = 0;
+    for (auto &&i: c->to_continue) {
+      auto j = pulling.find(i.hoid);
+      if (j != pulling.end()) {
+        auto rec_size = j->second.recovery_info.size;
+        auto rec_progress = j->second.recovery_progress.data_recovered_to;
+        cost += (rec_size - rec_progress);
```

Comment:
This is actually wrong.  PG_RecoveryQueueAsync here is going to enqueue a C_ReplicatedBackend_OnPullComplete which pushes the object data out to waiting acting set OSDs.  We want the full size of the relevant objects.  Instead, let's use i.stat.num_bytes_recovered for each element of to_continue.  See my branch.

Reply:
Understood. Thanks for clarifying this!